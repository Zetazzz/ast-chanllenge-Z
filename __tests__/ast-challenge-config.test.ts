import generate from "@babel/generator";
import queryASTCreator from "../src";
import methods from "../example-methods.json";

const expectCode = (ast) => {
  expect(generate(ast).code).toMatchSnapshot();
};

// test for cases on parameters
it("works for null input", () => {
  expectCode(queryASTCreator());
});

it("works for empty input", () => {
  expectCode(queryASTCreator({}));
});

it("works for combination of customized and default configs", () => {
  expectCode(
    queryASTCreator(
      {
        Pools: {
          requestType: "QueryPoolsRequest",
          responseType: "QueryPoolsResponse",
        },
      },
      {
        ___hookName___: "testUsePools",
      }
    )
  );
});

it("works for input request and response types override the types in configs", () => {
  expectCode(
    queryASTCreator(
      {
        Pools: {
          requestType: "InputQueryPoolsRequest",
          responseType: "InputQueryPoolsResponse",
        },
      },
      {
        ___requestType___: "ConfigQueryPoolsRequest",
      }
    )
  );
});
