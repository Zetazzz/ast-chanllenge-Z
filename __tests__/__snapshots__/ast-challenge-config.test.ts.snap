// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`works for combination of customized and default configs 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolsResponse, TData> {
  request?: QueryPoolsRequest;
}

const testUsePools = <TData = QueryPoolsResponse,>({
  request,
  options
}: UsePoolsQuery<TData>) => {
  return useQuery<QueryPoolsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pools(request);
  }, options);
};"
`;

exports[`works for empty input 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolsResponse, TData> {
  request?: QueryPoolsRequest;
}

const usePools = <TData = QueryPoolsResponse,>({
  request,
  options
}: UsePoolsQuery<TData>) => {
  return useQuery<QueryPoolsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pools(request);
  }, options);
};"
`;

exports[`works for input request and response types override the types in configs 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<InputQueryPoolsResponse, TData> {
  request?: InputQueryPoolsRequest;
}

const usePools = <TData = InputQueryPoolsResponse,>({
  request,
  options
}: UsePoolsQuery<TData>) => {
  return useQuery<InputQueryPoolsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pools(request);
  }, options);
};"
`;

exports[`works for null input 1`] = `
"export interface UsePoolsQuery<TData> extends ReactQueryParams<QueryPoolsResponse, TData> {
  request?: QueryPoolsRequest;
}

const usePools = <TData = QueryPoolsResponse,>({
  request,
  options
}: UsePoolsQuery<TData>) => {
  return useQuery<QueryPoolsResponse, Error, TData>([\\"poolsQuery\\", request], () => {
    if (!queryService) throw new Error(\\"Query Service not initialized\\");
    return queryService.pools(request);
  }, options);
};"
`;
